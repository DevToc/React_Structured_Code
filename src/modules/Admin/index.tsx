import {
  Alert,
  AlertIcon,
  Box,
  Button,
  Container,
  Flex,
  FormControl,
  FormHelperText,
  FormLabel,
  Heading,
  Input,
  Text,
  VStack,
} from '@chakra-ui/react';
import { ReactElement, useState } from 'react';
import { ReactComponent as VenngageLogoIcon } from 'assets/icons/venngage_logo.svg';
import { useLocalStorage } from 'hooks/useLocalStorage';
import { TemplateManager } from './TemplateManager';
import { InfographManager } from './InfographManager';
import { GeneratorPage } from './Generator';

interface AdminLoginProps {
  setJwtToken: (s: string) => void;
}

const AdminLogin = ({ setJwtToken }: AdminLoginProps): ReactElement => {
  const [tokenInput, setTokenInput] = useState<string>('');

  // TODO: Check the token before setting

  return (
    <Container mt={8}>
      <Alert status={'warning'}>
        <AlertIcon />
        Please enter JWT token to use admin services.
      </Alert>
      <FormControl mt={4}>
        <FormLabel>JWT Token</FormLabel>
        <Input value={tokenInput} type='text' onChange={(e) => setTokenInput(e.target.value)} />
        <FormHelperText>
          You are logging in as the JWT token user. Get this token from .. (explain where) This token will be saved in
          local storage.
        </FormHelperText>
      </FormControl>
      <Button mt={4} onClick={() => setJwtToken(tokenInput)}>
        Login
      </Button>
    </Container>
  );
};

const AdminDashboard = (): ReactElement => {
  return <Text>admin dashboard</Text>;
};

const AdminPage = (): ReactElement => {
  const [jwtToken, setJwtToken] = useLocalStorage<string>('jwtToken', '');
  const [section, setSection] = useState<'template_manager' | 'infograph_manager' | 'dashboard' | 'generator'>(
    'dashboard',
  );

  let SectionComponent;

  switch (section) {
    case 'dashboard':
      SectionComponent = <AdminDashboard />;
      break;
    case 'template_manager':
      SectionComponent = <TemplateManager jwtToken={jwtToken} />;
      break;
    case 'infograph_manager':
      SectionComponent = <InfographManager jwtToken={jwtToken} />;
      break;
    case 'generator':
      SectionComponent = <GeneratorPage jwtToken={jwtToken} />;
      break;
    default:
      SectionComponent = <Text>Invalid section</Text>;
  }

  return (
    <Box>
      <Flex p={2} justifyContent={'space-between'}>
        <Flex>
          <VenngageLogoIcon stroke='brand.500' />
          <Heading size={'lg'}>Admin</Heading>
        </Flex>
        {jwtToken !== '' ? (
          <Flex>
            <Button size={'sm'} onClick={() => setJwtToken('')}>
              Logout
            </Button>
          </Flex>
        ) : null}
      </Flex>
      {jwtToken !== '' ? (
        <Flex p={2}>
          <VStack w={'xs'} alignItems={'flex-start'}>
            <Button size={'sm'} onClick={() => setSection('dashboard')}>
              Dashboard
            </Button>
            <Button size={'sm'} onClick={() => setSection('template_manager')}>
              Templates
            </Button>
            <Button size={'sm'} onClick={() => setSection('infograph_manager')}>
              Infographs
            </Button>
            <Button size={'sm'} onClick={() => setSection('generator')}>
              Generator
            </Button>
          </VStack>
          {SectionComponent}
        </Flex>
      ) : (
        <AdminLogin setJwtToken={setJwtToken} />
      )}
    </Box>
  );
};

export default AdminPage;
