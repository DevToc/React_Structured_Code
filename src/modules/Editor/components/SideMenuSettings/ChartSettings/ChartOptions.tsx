import React, { memo, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import styled from '@emotion/styled';
import {
  ChakraComponent,
  Divider,
  Flex,
  FormControl,
  FormLabel,
  forwardRef,
  InputProps,
  NumberInput,
  NumberInputFieldProps,
  NumberInputProps,
  Radio,
  RadioGroup,
  Stack,
  Switch,
  Text,
} from '@chakra-ui/react';
import { ColorPicker } from '../../ColorPicker';
import { SettingsAccordion, SettingsAccordionToggle } from './ChartSettingsAccordion';
import {
  ChartAccordionProps,
  ChartSettingsColorPickerProps,
  ChartSliderInputProps,
  ChartSettingsTextProps,
  ChartSettingInputProps,
  ChartSettingsEntryProps,
  ChartSettingsRadioSelectorProps,
} from './ChartSettings.types';
import { SliderNumberInput } from 'modules/common/components/Input/SliderNumberInput';
import { NumberInputField } from 'modules/common/components/Input/NumberInputField';
import { Input } from 'modules/common/components/Input/Input';
import { useSideMenuSetting } from 'widgets/sdk';

/**
 * Shared settings menu components for all charts
 * Components can either be
 *
 * 1. Composed with the settings components:
 *  <VerticalAxis>
 *      <ChartSettingsColorPicker label='' color={color} onChange={onChange} />
 *      <SeriesColorPicker label='' color={color} onChange={onChange} />
 *  </VerticalAxis>
 *
 * 2. or custom components can be created with the base components: e.g.
 *   <SettingsAccordion title="custom component">
 *      <Heading>Custom</Heading>
 *      <MySlider />
 *      ...
 *   </SettingsAccordion>
 */
export const ChartSettingsEntry = memo(({ children, label, containerProps, labelProps }: ChartSettingsEntryProps) => {
  return (
    <Flex alignItems='center' gap={3} justifyContent='space-between' color='font.500' fontSize='sm' {...containerProps}>
      <Text margin={0} fontSize='sm' fontWeight={'normal'} {...labelProps}>
        {label}
      </Text>
      <Flex alignItems='right' gap={3}>
        {children}
      </Flex>
    </Flex>
  );
});

export const TextSetting: ChakraComponent<typeof Input, ChartSettingInputProps & InputProps> = forwardRef(
  function TextSetting(inProps, ref) {
    const { label, testId, containerProps, labelProps, ...otherProps } = inProps;

    return (
      <FormControl as={Flex} align='center' justifyContent={'space-between'} {...containerProps}>
        <FormLabel margin={0} fontSize='sm' fontWeight={'normal'} {...labelProps}>
          {label}
        </FormLabel>
        <Input
          ref={ref}
          size={'xs'}
          padding={'4px 8px'}
          margin={0}
          borderRadius={4}
          width={'3xs'}
          data-testid={testId}
          borderColor='outline.gray'
          {...otherProps}
        />
      </FormControl>
    );
  },
);

export const NumberSetting: ChakraComponent<
  typeof NumberInput,
  ChartSettingInputProps & NumberInputProps & { inputProps?: NumberInputFieldProps }
> = forwardRef(function NumberSetting(inProps, ref) {
  const { label, testId, containerProps, labelProps, inputProps, placeholder, ...otherProps } = inProps;
  return (
    <FormControl as={Flex} align='center' justifyContent={'space-between'} {...containerProps}>
      <FormLabel margin={0} fontSize='sm' fontWeight={'normal'} {...labelProps}>
        {label}
      </FormLabel>
      <NumberInput size={'xs'} {...otherProps}>
        <NumberInputField
          ref={ref}
          placeholder={placeholder}
          data-testid={testId}
          borderColor='outline.gray'
          {...inputProps}
        />
      </NumberInput>
    </FormControl>
  );
});

export const Legend = memo(
  ({ children, onToggle, isChecked }: { children: React.ReactNode; onToggle: () => void; isChecked: boolean }) => {
    const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false });
    const title = t('chartSettings.legendTitle');

    return (
      <SettingsAccordionToggle onToggle={onToggle} isChecked={isChecked} title={title}>
        {children}
      </SettingsAccordionToggle>
    );
  },
);

export const Switcher = memo(
  ({ onToggle, isChecked, label }: { onToggle: () => void; isChecked: boolean; label: string }) => {
    const switchId = label.replaceAll(' ', '-').toLowerCase() + 'toggle';

    return (
      <Flex
        alignItems='center'
        gap={3}
        justifyContent='space-between'
        color='font.500'
        mr='auto'
        fontSize='sm'
        py='2'
        mt='2'
      >
        <FormControl display='flex' alignItems='center'>
          <FormLabel htmlFor={switchId} mr='auto' fontSize='sm' fontWeight='400' mb='0'>
            {label}
          </FormLabel>
          <Switch onChange={onToggle} isChecked={isChecked} id={switchId} />
        </FormControl>
      </Flex>
    );
  },
);

export const NumberFormat = memo(({ children }: ChartAccordionProps) => {
  const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false });
  const title = t('chartSettings.numberFormat');

  return (
    <SettingsAccordion title={title} isDefaultOpen>
      {children}
    </SettingsAccordion>
  );
});

export const Axis = memo(({ children }: { children: React.ReactNode }): ReactElement => {
  // const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false }); // TODO: on new PR
  const title = 'Axes';

  return (
    <SettingsAccordion
      title={title}
      accordionPanelProps={{
        display: 'flex',
        gap: 3,
        flexDirection: 'column',
      }}
    >
      {children}
    </SettingsAccordion>
  );
});

export const NumberFormatSwitch = memo(function NumberFormatSwitch({
  onChange,
  value,
}: ChartSettingsRadioSelectorProps<'percentage' | 'value'>) {
  const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false });
  const label = t('chartSettings.numberFormatFormat');
  const percentageText = t('chartSettings.numberFormatPercentage');
  const valueText = t('chartSettings.numberFormatValue');

  return (
    <FormControl id={'data-table-format'} as={Flex} align={'center'} marginBottom={1}>
      <Text id={'data-table-format-label'} mr='auto' fontSize={'sm'}>
        {label}
      </Text>
      <RadioGroup onChange={onChange} value={value} name={label}>
        <Stack direction={'row'} spacing={6}>
          <Radio id={'data-format-percentage'} value={'percentage'} size={'sm'}>
            {percentageText}
          </Radio>
          <Radio id={'data-format-value'} value={'value'} size={'sm'}>
            {valueText}
          </Radio>
        </Stack>
      </RadioGroup>
    </FormControl>
  );
});

export const DataPrefix = memo(({ value, onBlur, onChange }: ChartSettingsTextProps) => {
  const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false });
  const label = t('chartSettings.dataPrefix');

  return (
    <Flex align='center'>
      <Text mr='auto' fontSize='sm'>
        {label}
      </Text>
      <Input
        size={'xs'}
        fontSize={'12px'}
        padding={'6px 8px 6px 8px'}
        margin={'6px 0'}
        borderRadius={4}
        borderColor={'outline.gray'}
        width={'7rem'}
        aria-label='number format prefix'
        placeholder='e.g. $, £, €'
        value={value}
        onBlur={onBlur}
        onChange={onChange}
        data-testid='number-format-prefix'
      />
    </Flex>
  );
});

export const DataSuffix = memo(({ value, onBlur, onChange }: ChartSettingsTextProps) => {
  const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false });
  const label = t('chartSettings.dataSuffix');

  return (
    <Flex align='center'>
      <Text mr='auto' fontSize='sm'>
        {label}
      </Text>
      <Input
        size={'xs'}
        fontSize={'12px'}
        padding={'6px 8px 6px 8px'}
        margin={'8px 0'}
        borderRadius={4}
        borderColor={'outline.gray'}
        width={'7rem'}
        aria-label='number format suffix'
        placeholder='e.g. kg, ft, M, °'
        value={value}
        onBlur={onBlur}
        onChange={onChange}
        data-testid='number-format-suffix'
      />
    </Flex>
  );
});

export const Labels = memo(({ children, isCheck, onToggle = () => {} }: ChartAccordionProps) => {
  const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false });
  const title = t('chartSettings.labelsTitle');

  // if the enable labels toggle is not undefined, which is a boolean,
  // then we need to have a toggle button on the accordion header.
  // Therefore, render accordion component with a toggle instead of the normal setting accordion.
  const Component = typeof isCheck === 'boolean' ? SettingsAccordionToggle : SettingsAccordion;

  return (
    <Component
      title={title}
      isChecked={isCheck ?? false} // When isChecked is undefined, isCheck is not used. fallback to false for typecheck
      onToggle={onToggle}
      accordionPanelProps={{
        display: 'flex',
        gap: 3,
        flexDirection: 'column',
      }}
    >
      {children}
    </Component>
  );
});

export const VerticalAxis = memo(({ isDefaultOpen, children }: ChartAccordionProps) => {
  const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false });
  const title = t('chartSettings.verticalAxisTitle');

  return <SettingsAccordion title={title}>{children}</SettingsAccordion>;
});

export const HorizontalAxis = memo(({ isDefaultOpen, children }: ChartAccordionProps) => {
  const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false });
  const title = t('chartSettings.horizontalAxisTitle');

  return <SettingsAccordion title={title}>{children}</SettingsAccordion>;
});

export const GridLines = memo(({ isDefaultOpen, children }: ChartAccordionProps) => {
  const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false });
  const title = t('chartSettings.gridLinesTitle');

  return <SettingsAccordion title={title}>{children}</SettingsAccordion>;
});

export const ChartStyle = ({ isDefaultOpen, children }: ChartAccordionProps) => {
  const { t } = useTranslation('editor_side_menu_settings', { useSuspense: false });
  const title = t('chartSettings.chartStyleTitle');

  return (
    <SettingsAccordion isDefaultOpen={isDefaultOpen} title={title}>
      {children}
    </SettingsAccordion>
  );
};

export const ChartSettingsColorPicker = memo(({ label, color, onChange }: ChartSettingsColorPickerProps) => {
  const { setAllowScrolling } = useSideMenuSetting();
  return (
    <Flex align='center'>
      <Text mr='auto' fontSize='sm'>
        {label}
      </Text>
      <ColorPicker
        color={color}
        onChange={onChange}
        onClose={setAllowScrolling.on}
        onOpen={setAllowScrolling.off}
        placement='left'
      />
    </Flex>
  );
});

export const ChartSliderInput = ({ title, value, onChange, minValue, maxValue, suffix }: ChartSliderInputProps) => {
  return (
    <SliderNumberInput
      title={title}
      value={value}
      minValue={minValue}
      maxValue={maxValue}
      suffix={suffix}
      fontSize={'sm'}
      titleProps={{ fontWeight: 'normal' }}
      numberInputProps={{ textAlign: 'left', w: 12 }}
      sliderWidth={120}
      onChange={onChange}
      containerProps={{ pt: 2 }}
    />
  );
};

export const ChartSettingsDivider = styled(Divider)`
  width: calc(100% + 2rem);
  margin: 1rem 0;
  margin-left: -1rem;
`;
