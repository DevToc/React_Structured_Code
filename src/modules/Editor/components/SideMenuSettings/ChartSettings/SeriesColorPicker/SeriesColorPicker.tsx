import { memo } from 'react';
import { SeriesColorPickerProps } from './SeriesColorPicker.types';
import { Flex, Text } from '@chakra-ui/react';
import { ColorPicker } from 'modules/Editor/components/ColorPicker';
import { useSideMenuSetting } from 'widgets/sdk';

const SeriesColorPicker = memo(({ label, series, placement = 'left-start', onChange }: SeriesColorPickerProps) => {
  const { setAllowScrolling } = useSideMenuSetting();
  return (
    <>
      <Text mb={2} fontSize='sm'>
        {label}
      </Text>
      {series.map(({ color, name }, index: number) => {
        const onChangeDataColor = (color: string) => onChange(color, index);

        return (
          <Flex align='center' key={index}>
            <ColorPicker
              color={color}
              onChange={onChangeDataColor}
              onOpen={setAllowScrolling.off}
              onClose={setAllowScrolling.on}
              placement={placement}
            />
            <Text pl={3} fontSize='sm'>
              {name}
            </Text>
          </Flex>
        );
      })}
    </>
  );
});

export default SeriesColorPicker;
