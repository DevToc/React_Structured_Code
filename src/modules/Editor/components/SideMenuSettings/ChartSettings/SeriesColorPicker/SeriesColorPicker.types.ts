import { PlacementWithLogical } from '@chakra-ui/react';

type Color = string;
type Name = string;

export type Series = { color: Color; name: Name };

export interface SeriesColorPickerProps {
  label: string;
  series: Series[];
  onChange: (color: string, index: number) => void;

  placement?: PlacementWithLogical;
}
