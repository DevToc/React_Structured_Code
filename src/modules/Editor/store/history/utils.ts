export const isEmpty = (value: any): boolean => {
  return (
    value === null ||
    value === undefined ||
    (typeof value === 'object' && !Object.values(value).some((v) => v !== null && typeof v !== 'undefined')) ||
    (typeof value === 'string' && value.length === 0)
  );
};

export const isNotEmpty = (value: any) => !isEmpty(value);
