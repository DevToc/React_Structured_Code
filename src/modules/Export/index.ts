import Export from './Export';
export default Export;
