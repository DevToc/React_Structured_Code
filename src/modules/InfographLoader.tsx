import { ReactNode, ReactElement, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useAppDispatch, useAppSelector, setActivePage } from './Editor/store';
import { setFontsLoaded } from './Editor/store/pageControlSlice';
import { filterUnloadedFamilies, loadFonts } from '../hooks/useFont';
import { findFontsInPage } from './Editor/store/infographSlice.helpers';
import { loadInfograph } from './Editor/store/infographSlice';
import widgetVersionManager from '../widgets/WidgetVersionManager';
import { getFullInfograph } from '../libs/e2api/get';
import { InfographState } from '../types/infographTypes';
import { setOriginalEditorId } from './Editor/store/editorSettingsSlice';
import { removeDeletedWidgets } from '../widgets/Widget.helpers';
import { selectActivePage } from './Editor/store/pageSelector';
import { selectInfograph } from './Editor/store/infographSelector';

interface InfographLoaderProps {
  children: ReactNode;
  infographState?: InfographState;
}

/**
 * Loads infograph to the store.
 *
 * If infographId param is available in the route, loads it from the backend api.
 *
 * If static local infograph is provided in prop, it'll load it.
 *
 * @param children
 * @param infographState
 * @constructor
 */
const InfographLoader = ({ children, infographState }: InfographLoaderProps): ReactElement => {
  const params = useParams();

  const { id: infographId, pageId, privateLinkId } = params || {};

  const dispatch = useAppDispatch();
  const infograph = useAppSelector(selectInfograph);
  const activePageId = useAppSelector(selectActivePage);

  useEffect(() => {
    const fetchId = infographId ?? privateLinkId;
    if (fetchId) {
      // Fetch infograph
      getFullInfograph(fetchId, !!privateLinkId).then((infograph) => {
        // console.debug('fetched infograph', infograph)
        const openPageId = pageId || infograph?.pageOrder[0] || '';

        // Load fonts asynchronously, to block page render before font fetching, use await loadFonts instead
        // TODO: hmm clean this up into single dispatch?
        // Set e1 uuid as a reference id
        !!infographId && dispatch(setOriginalEditorId(infographId));

        // Temporary: filter out deleted widgets from the Infograph
        // until they are removed from the infograph object returned from the BE (as they are not updated)
        const filteredInfograph = removeDeletedWidgets(infograph);

        // update widgets to latest version
        const updatedWidgets = widgetVersionManager.update(filteredInfograph.widgets);
        filteredInfograph.widgets = updatedWidgets;

        dispatch(loadInfograph(filteredInfograph));
        dispatch(setActivePage({ pageId: openPageId }));
        loadFonts(findFontsInPage(filteredInfograph, openPageId)).then(() => {
          dispatch(setFontsLoaded(true));
        });
      });
    } else if (infographState) {
      const pageId = infographState.pageOrder[0] || '';
      dispatch(loadInfograph(infographState));
      dispatch(setActivePage({ pageId }));
      //   TODO: improve this, not loading fonts here as this is used for integration tests
      //   loadFonts(findFontsInPage(infograph, openPageId));
    }
  }, [dispatch, infographId, infographState, pageId, privateLinkId]);

  useEffect(() => {
    // Handle multipage font loads
    // when the current active page changes
    // Skip the first page load
    if (!!activePageId && infograph?.pageOrder.length > 1 && infograph?.pageOrder[0] !== activePageId) {
      const fonts = findFontsInPage(infograph, activePageId);
      const unloadedFamilies = filterUnloadedFamilies(fonts);

      if (unloadedFamilies.length > 0) {
        dispatch(setFontsLoaded(false));
        loadFonts(unloadedFamilies).then(() => {
          dispatch(setFontsLoaded(true));
        });
      }
    }
  }, [dispatch, infograph, activePageId]);

  return <>{children}</>;
};

export { InfographLoader };
