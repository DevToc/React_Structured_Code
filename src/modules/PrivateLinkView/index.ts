import PrivateLinkView from './PrivateLinkView';
export default PrivateLinkView;
