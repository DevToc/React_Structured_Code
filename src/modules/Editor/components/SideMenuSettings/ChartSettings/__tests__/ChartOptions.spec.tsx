import { screen } from '@testing-library/react';
import { InfographLoader } from 'modules/InfographLoader';
import { renderWithRedux } from 'utils/test-utils.test';
import { EMPTY_INFOGRAPH } from 'utils/loadSampleInfograph';
import { DataPrefix, DataSuffix, NumberFormat, NumberFormatSwitch } from '../ChartOptions';
import SeriesColorPicker from '../SeriesColorPicker/SeriesColorPicker';
import { SideMenuSettingProvider } from 'widgets/sdk';

describe('ChartOptions.tsx', () => {
  global.window.scrollTo = jest.fn();

  it('should render SeriesColorPicker labels', () => {
    const { container } = renderWithRedux(
      <InfographLoader infographState={EMPTY_INFOGRAPH}>
        <SideMenuSettingProvider>
          <SeriesColorPicker
            label='SeriesColorPicker Label'
            series={[
              { color: '#fff', name: 'Apples' },
              { color: '#000', name: 'Oranges' },
            ]}
            onChange={() => {}}
          />
        </SideMenuSettingProvider>
      </InfographLoader>,
    );

    expect(container).toBeInTheDocument();
    expect(screen.getByText('SeriesColorPicker Label')).toBeInTheDocument();
    expect(screen.getByText('Apples')).toBeInTheDocument();
    expect(screen.getByText('Oranges')).toBeInTheDocument();
  });

  it('should render DataPrefix', () => {
    const { asFragment } = renderWithRedux(<DataPrefix value='Test' onChange={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render DataSuffix', () => {
    const { asFragment } = renderWithRedux(<DataSuffix value='Test' onChange={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render NumberFormat', () => {
    const { asFragment } = renderWithRedux(
      <NumberFormat>
        <DataPrefix value='Test' onChange={() => {}} />
      </NumberFormat>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render NumberFormatSwitch (percentage)', () => {
    const { asFragment } = renderWithRedux(<NumberFormatSwitch onChange={() => {}} value={'percentage'} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render NumberFormatSwitch (value)', () => {
    const { asFragment } = renderWithRedux(<NumberFormatSwitch onChange={() => {}} value={'percentage'} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
