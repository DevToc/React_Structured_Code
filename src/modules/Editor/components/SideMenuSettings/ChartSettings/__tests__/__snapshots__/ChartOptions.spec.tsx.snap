// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartOptions.tsx should render DataPrefix 1`] = `
<DocumentFragment>
  <div
    class="css-70qvj9"
  >
    <p
      class="chakra-text css-1mtakt3"
    >
      Prefix
    </p>
    <input
      aria-label="number format prefix"
      class="chakra-input css-1fswsip"
      data-testid="number-format-prefix"
      placeholder="e.g. $, £, €"
      value="Test"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`ChartOptions.tsx should render DataSuffix 1`] = `
<DocumentFragment>
  <div
    class="css-70qvj9"
  >
    <p
      class="chakra-text css-1mtakt3"
    >
      Suffix
    </p>
    <input
      aria-label="number format suffix"
      class="chakra-input css-1j0he9p"
      data-testid="number-format-suffix"
      placeholder="e.g. kg, ft, M, °"
      value="Test"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`ChartOptions.tsx should render NumberFormat 1`] = `
<DocumentFragment>
  <div
    class="chakra-accordion css-x7re8g"
  >
    <div
      class="chakra-accordion__item css-1pfn623"
    >
      <h2>
        <button
          aria-controls="accordion-panel-:r8:"
          aria-expanded="true"
          class="chakra-accordion__button css-1oc1f0x"
          data-index="0"
          id="accordion-button-:r8:"
          type="button"
        >
          <div
            class="css-nfa4jm"
          >
            Number Format
          </div>
          <svg
            aria-hidden="true"
            class="chakra-icon chakra-accordion__icon css-j2ph2z"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h2>
      <div
        class="chakra-collapse"
        style="overflow: hidden; display: block; opacity: 0; height: auto;"
      >
        <div
          aria-labelledby="accordion-button-:r8:"
          class="chakra-accordion__panel css-rv0asf"
          id="accordion-panel-:r8:"
          role="region"
        >
          <div
            class="css-70qvj9"
          >
            <p
              class="chakra-text css-1mtakt3"
            >
              Prefix
            </p>
            <input
              aria-label="number format prefix"
              class="chakra-input css-1fswsip"
              data-testid="number-format-prefix"
              placeholder="e.g. $, £, €"
              value="Test"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`ChartOptions.tsx should render NumberFormatSwitch (percentage) 1`] = `
<DocumentFragment>
  <div
    class="chakra-form-control css-1wth8of"
    role="group"
  >
    <p
      class="chakra-text css-1mtakt3"
      id="data-table-format-label"
    >
      Format
    </p>
    <div
      class="chakra-radio-group css-0"
      role="radiogroup"
    >
      <div
        class="chakra-stack css-mc07to"
      >
        <label
          class="chakra-radio css-1cqh9jq"
          data-checked=""
        >
          <input
            checked=""
            class="chakra-radio__input"
            id="data-format-percentage"
            name="Format"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="radio"
            value="percentage"
          />
          <span
            aria-hidden="true"
            class="chakra-radio__control css-arfnvs"
            data-checked=""
          />
          <span
            class="chakra-radio__label css-lest1k"
            data-checked=""
          >
            Percentage
          </span>
        </label>
        <label
          class="chakra-radio css-1cqh9jq"
        >
          <input
            class="chakra-radio__input"
            id="data-format-value"
            name="Format"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="radio"
            value="value"
          />
          <span
            aria-hidden="true"
            class="chakra-radio__control css-arfnvs"
          />
          <span
            class="chakra-radio__label css-lest1k"
          >
            Value
          </span>
        </label>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`ChartOptions.tsx should render NumberFormatSwitch (value) 1`] = `
<DocumentFragment>
  <div
    class="chakra-form-control css-1wth8of"
    role="group"
  >
    <p
      class="chakra-text css-1mtakt3"
      id="data-table-format-label"
    >
      Format
    </p>
    <div
      class="chakra-radio-group css-0"
      role="radiogroup"
    >
      <div
        class="chakra-stack css-mc07to"
      >
        <label
          class="chakra-radio css-1cqh9jq"
          data-checked=""
        >
          <input
            checked=""
            class="chakra-radio__input"
            id="data-format-percentage"
            name="Format"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="radio"
            value="percentage"
          />
          <span
            aria-hidden="true"
            class="chakra-radio__control css-arfnvs"
            data-checked=""
          />
          <span
            class="chakra-radio__label css-lest1k"
            data-checked=""
          >
            Percentage
          </span>
        </label>
        <label
          class="chakra-radio css-1cqh9jq"
        >
          <input
            class="chakra-radio__input"
            id="data-format-value"
            name="Format"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="radio"
            value="value"
          />
          <span
            aria-hidden="true"
            class="chakra-radio__control css-arfnvs"
          />
          <span
            class="chakra-radio__label css-lest1k"
          >
            Value
          </span>
        </label>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
